src/main/java/randoop/reflection/SignatureParser.java

```diff
@@ -0,0 +1,154 @@
+package randoop.reflection;
+
+import java.lang.reflect.AccessibleObject;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Method;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/** Parses type signature strings used to identify methods and constructors in input. */
+// TODO: This duplicates should be factored into a separate source set (aka, module) so that it
```

Comment:
it is probably ~1/2 day, but it would be easier to do once everything is merged, because it will affect the build.